### PR TITLE
Fills out IProjectRepository

### LINKED ISSUE #105: Fill out IProjectRepository

### PR DESCRIPTION
**Pull request by EmmKirHan**
Fixes #105

### CODE REVIEW

File: ProjectBank/Core/IProjectRepository.cs [hunk 1]
```diff
@@ -1,5 +1,10 @@
 namespace ProjectBank.Core;
-
-public class IProjectRepository
+public interface IProjectRepository
 {
+    Task<ProjectDTO> CreateProjectAsync(ProjectCreateDTO project);
+    Task<IReadOnlyCollection<ProjectDTO>> ReadAllAsync();
+    Task<ProjectDTO> ReadProjectAsync(int projectId);
+    Task<IReadOnlyCollection<ProjectDTO>> ReadProjectsByUserAsync(int userId);
+    Task<Response> UpdateProjectAsync(ProjectDTO project);
```
Comment (Werenfeldt) [inline ProjectBank/Core/IProjectRepository.cs:8]: Skal denne måske deles op i to? 
Én til når en supervisor "EditProjectAsync(ProjectDTO project)", og Én til når project ændre status fra Active til Closed
"UpdateStatusProjectAsync(int projectId)"

**Review by Werenfeldt — CHANGES_REQUESTED** [review]

File: ProjectBank/Core/IProjectRepository.cs (hunk 1 above)
Comment (JeppeKorgaard98) [reply ProjectBank/Core/IProjectRepository.cs:8]: Jeg tror også vi skal have en der returnerer alle projects med bestemt tag

**Review by frejabv — COMMENTED** [review]
Why are so many files changed? Does it get fixed if you merge main into this branch?

**Review by JeppeKorgaard98 — APPROVED** [review]

**Review by Werenfeldt — APPROVED** [review]

File: ProjectBank/Core/IProjectRepository.cs [hunk 2]
```diff
@@ -1,5 +1,12 @@
 namespace ProjectBank.Core;
-
-public class IProjectRepository
+public interface IProjectRepository
 {
+    Task<ProjectDTO> CreateProjectAsync(ProjectCreateDTO project);
+    Task<IReadOnlyCollection<ProjectDTO>> ReadAllAsync();
+    Task<ProjectDTO> ReadProjectAsync(int projectId);
```
Comment (JacobMoller) [inline ProjectBank/Core/IProjectRepository.cs:6]: This should be renamed to "ReadProjectByIdAsync" to be in line with the other methods that takes a specific id.

File: ProjectBank/Core/IProjectRepository.cs [hunk 3]
```diff
@@ -1,5 +1,12 @@
...
+    Task<IReadOnlyCollection<ProjectDTO>> ReadProjectsByUserAsync(int userId);
+    Task<IReadOnlyCollection<ProjectDTO>> ReadProjectsByTag(int tagId);
```
Comment (JacobMoller) [inline ProjectBank/Core/IProjectRepository.cs:8]: This should be renamed to "ReadProjectsByTagAsync" to reflect that it is a async method

File: ProjectBank/Core/IProjectRepository.cs [hunk 4]
```diff
@@ -1,5 +1,12 @@
...
+    Task<Response> EditProjectAsync(ProjectDTO project);
+    Task<Response> UpdateProjectStatusAsync(int projectId);
```
Comment (JacobMoller) [inline ProjectBank/Core/IProjectRepository.cs:10]: This should be renamed to "UpdateProjectStatusByIdAsync"

File: ProjectBank/Core/IProjectRepository.cs [hunk 5]
```diff
@@ -1,5 +1,12 @@
...
+    Task<Response> DeleteProjectAsync(int projectId);
```
Comment (JacobMoller) [inline ProjectBank/Core/IProjectRepository.cs:11]: This should be renamed to "DeleteProjectByIdAsync"

**Review by JacobMoller — CHANGES_REQUESTED** [review]

**Review by JacobMoller — APPROVED** [review]

**Pull request merged**